File: rectangle.cpp
//------------------------------------------------------------------------------
// rectangle_In.cpp - содержит процедуру ввода параметров 
// для уже созданного прямоугольника
//------------------------------------------------------------------------------

#include "rectangle.h"
#include "shape.h"

//------------------------------------------------------------------------------
// Ввод параметров прямоугольника из файла
void In(rectangle &r, ifstream &ifst) {
    int color_code;
    ifst >> r.x >> r.y >> r.x1 >> r.y1 >> color_code;
    r.col = colors[color_code];
}

// Случайный ввод параметров прямоугольника
void InRnd(rectangle &r) {
    r.x = Random();
    r.y = Random();
    r.x1 = Random();
    do {
        r.y1 = Random();
    } while (r.y == r.y1);
    r.col = colors[Random() % 7];
}

//------------------------------------------------------------------------------
// Вывод параметров прямоугольника в форматируемый поток
void Out(rectangle &r, std::ofstream &ofst) {
    ofst << "It is Rectangle: x = (" << r.x << ", " << r.y << ") "
         << "x_1 = (" << r.x1 << ", " << r.y1 << ")"
         << ". Area = " << Area(r)
         << ". Color = " << GetColor(r) << "\n";
}

//------------------------------------------------------------------------------
// Вычисление периметра прямоугольника
double Area(rectangle &r) {
    return Length(r.x, r.y, r.x, r.y1) * Length(r.x, r.y, r.x1, r.y);
}

color GetColor(rectangle& r) {
    return r.col;
}

File: figures/rectangle.py
import random
import helper


class Rectangle:
    def __init__(self, x_=random.randint(1, 100), y_=random.randint(1, 100), x1_=random.randint(1, 100),
                 y1_=random.randint(1, 100), color_code_=random.randint(0, 6)):
        self.name = 'Rectangle'
        self.x = x_
        self.y = y_
        self.y1 = y1_
        self.x1 = x1_
        self.color_code = color_code_

        while self.area() == 0:
            self.x1 = random.randint(1, 100)

        self.calculated_area = self.area()

    def area(self):
        return helper.length(self.x, self.y, self.x1, self.y) * helper.length(self.x, self.y, self.x, self.y1)

    def toString(self):
        return f'{self.name} X: {self.x} Y: {self.y} X1: {self.x1} Y1: {self.y1} S: {self.calculated_area} ' \
               f'Color: {helper.getColorNameByCode(self.color_code)}'
